DocData and virtual method type hints fixes

neikeq opened the pull request:
- `DocData` and type hints fixes:
  - Makes vararg methods automatically use `PROPERTY_USAGE_NIL_IS_VARIANT` on return types
  - Completely removes the `:type` suffix for method names. Virtual methods must use the `MethodInfo` constructors that takes `Variant::Type` or `PropertyHint` as the first parameter for the return type (with `CLASS_INFO` as a helper to get the `PropertyInfo`). Parameters must use `PROPERTY_HINT_RESOURCE_TYPE` and hint string.
  - `PROPERTY_USAGE_NIL_IS_VARIANT` is no longer needed for parameters, because parameters cannot be void.
  - Adds missing `PROPERTY_USAGE_NIL_IS_VARIANT` to virtual and built-in methods that return Variant.
- Fixes missing default arguments in `MethodInfo`.
- Makes built-in vararg methods actual vararg methods:
  - Removes hardcoded parameters from built-in vararg methods and adds `METHOD_FLAG_VARARG` to them.
  - Makes `EditorHelp` display built-in vararg methods correctly

neikeq committed:
DocData and type hints fixes
- Makes vararg methods automatically use PROPERTY_USAGE_NIL_IS_VARIANT on return types
- Completely removes the ":type" suffix for method names. Virtual methods must use the MethodInfo constructors that takes Variant::Type or PropertyHint as the first parameter for the return type (with CLASS_INFO as a helper to get the PropertyInfo). Parameters must use PROPERTY_HINT_RESOURCE_TYPE and hint string.
- PROPERTY_USAGE_NIL_IS_VARIANT is no longer needed for parameters, because parameters cannot be void.
- Adds missing PROPERTY_USAGE_NIL_IS_VARIANT to virtual and built-in methods that return Variant.

akien-mga commented:
It fails building with clang: https://travis-ci.org/godotengine/godot/jobs/269450949
```
core/object.h:165:5: error: delegating constructors are permitted only in C++11
                : PropertyInfo(p_type, p_name, PROPERTY_HINT_NONE, "", p_usage) {
                  ^~~~~~~~~~~~
```

neikeq commented:
Oh, shit.

neikeq committed:
Fixes missing default arguments in MethodInfo

neikeq committed:
Makes built-in vararg methods actual vararg methods
- Removes hardcoded parameters from built-in vararg methods and adds METHOD_FLAG_VARARG to them.
- Makes EditorHelp display built-in vararg methods correctly.

neikeq commented:
All right, this is ready now.
